### PR TITLE
use zap for reserve-data

### PR DESCRIPTION
**Pull request by secmask**

### CODE REVIEW

**Comment by secmask** [comment]
> * Should we pass `zap.SugarLogger` as params or using `zap.S()` to get a global sugar logger in order to reduce the number of params and minimum the changes in code base
> * Should we only use Debugw Warnw Errorw only and remove Debugf Warnf Errorf

1) yeah, I thought about that at first too, but ending with not to use global thing.
2) `w` a bit faster but it require to re-make message sentence to make sense, so I chose the quick way.

**Comment by secmask** [comment]
I updated to use zap global sugar, so it would easier for us later apply on v1.
Most log statement under err context will be use Warn level, I need your suggest on which situation to use Error level.

File: common/blockchain/base_blockchain.go [hunk 1]
```diff
@@ -107,9 +108,9 @@ func (b *BaseBlockchain) SignAndBroadcast(tx *types.Transaction, from string) (*
 		return nil, err
 	}
 	failures, ok := b.broadcaster.Broadcast(signedTx)
-	log.Printf("Rebroadcasting failures: %s", failures)
+	b.l.Infof("Rebroadcasting failures: %s", failures)
 	if !ok {
-		log.Printf("Broadcasting transaction failed! nonce: %d, gas price: %s, retry failures: %s", tx.Nonce(), tx.GasPrice().Text(10), failures)
+		b.l.Warnf("Broadcasting transaction failed! nonce: %d, gas price: %s, retry failures: %s", tx.Nonce(), tx.GasPrice().Text(10), failures)
```
Comment (ducquangkstn) [inline common/blockchain/base_blockchain.go]: adding tx_hash make debug easier in the future

File: data/fetcher/fetcher.go [hunk 2]
```diff
@@ -73,62 +75,62 @@ func (f *Fetcher) Run() error {
...
 	go f.RunFetchExchangeHistory()
-	log.Printf("Fetcher runner is running...")
+	f.l.Infof("Fetcher runner is running...")
 	return nil
 }
 
 func (f *Fetcher) RunGlobalDataFetcher() {
 	for {
-		log.Printf("waiting for signal from global data channel")
+		f.l.Info("waiting for signal from global data channel")
```
[the remark this answers is not in the document]
Comment (secmask) [reply data/fetcher/fetcher.go]: why we want a warn here?

File: http/server.go [hunk 3]
```diff
@@ -32,32 +33,33 @@ type Server struct {
 	r              *gin.Engine
 	blockchain     Blockchain
 	settingStorage storage.Interface
+	l              *zap.SugaredLogger
 }
 
-func getTimePoint(c *gin.Context, useDefault bool) uint64 {
+func getTimePoint(c *gin.Context, useDefault bool, l *zap.SugaredLogger) uint64 {
```
[the remark this answers is not in the document]
Comment (secmask) [reply http/server.go]: all right.

File: common/blockchain/base_blockchain.go (hunk 1 above)
Comment (secmask) [reply common/blockchain/base_blockchain.go]: 👍

File: data/fetcher/fetcher.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ducquangkstn) [reply data/fetcher/fetcher.go]: Sr, just ignore thjs

**Pull request merged**